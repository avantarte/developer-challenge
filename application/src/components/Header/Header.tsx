import React from "react";
import logo from "../../styles/logo.svg";
import styled from "styled-components";

const StyledHeader = styled.header`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: white;

  box-shadow: ${(props) => props.theme.header.boxShadowSm.boxShadow};
`;

const StyledHeaderTitle = styled.span`
  ${(props) => props.theme.header.visuallyHidden}
`;

const StyledHeaderImage = styled.img`
  height: 35px;
  padding: 10px;
  vertical-align: top;
`;

const Header: React.FC = () => {
  return (
    <StyledHeader>
      <a href="/">
        <StyledHeaderTitle>Avant Arte</StyledHeaderTitle>
        <StyledHeaderImage alt="AvantArte" src={logo} />
      </a>
    </StyledHeader>
  );
};

export default Header;
