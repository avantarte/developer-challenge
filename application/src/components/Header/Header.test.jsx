import Header from "./Header";
import React from "react";
import TestRenderer from "react-test-renderer";

describe("Header", () => {
  it("should render correctly", () => {
    const tree = TestRenderer.create(<Header />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
