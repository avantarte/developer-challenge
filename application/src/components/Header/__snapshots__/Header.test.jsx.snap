// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<header
  className="Header"
>
  <a
    href="/"
  >
    <span
      className="Header__hidden-text"
    >
      Avant Arte
    </span>
    <img
      alt="AvantArte"
      className="Header__logo-image"
      src="logo.svg"
    />
  </a>
</header>
`;
